### ISSUE TITLE
Partition by should not quote the items

### ISSUE THREAD

**Issue by Conq1** [dremio/dbt-dremio#201]
### Describe the enhancement requested

I would like to use the features of Iceberg partition transformation, but when adding them to "partition_by" in config they get quotes surrounding each item, meaning the create table will fail.

### Justification for this enhancement

Partition transformation is a powerful Iceberg feature.

**Comment by Conq1** [comment]
For now I have created my own macro of config_cols. 
Might be a better solution as config_cols is used multible other places.

New version without quoting:
```
{% macro config_cols(label, default_cols=none) %}
  {%- set cols = config.get(label | replace(" ", "_"), validator=validation.any[list, basestring]) or default_cols -%}
  {%- if cols is not none %}
    {%- if cols is string -%}
      {%- set cols = [cols] -%}
    {%- endif -%}
    {{ label }} (
    {{ cols | join(', ') }}
    )
  {%- endif %}
{%- endmacro -%}

```

**Comment by fabrice-etanchaud** [comment]
Hi @Conq1 , dremio usually accepts double quotes around object names (I just tested a partitioned table creation with partition by ("mycol")), are you sure the error is not coming from another problem, maybe a misnamed column ?

**Comment by Conq1** [comment]
Yes @fabrice-etanchaud . Problem is when you want to use partition transformation like TRUNC(1, "mycol"):
https://docs.dremio.com/software/sql-reference/sql-commands/apache-iceberg-tables/apache-iceberg-create/
So quick fix for me was to remove the quotes, but a better fix maybe some other syntax to indicate when you want to use partition transformation.

**Comment by fabrice-etanchaud** [comment]
Thank you @Conq1 , I understand. You are right, the solution may be to remove the systematic double quoting, and let the user decide.

**Comment by wabu** [comment]
I'm facing the same issue. Any progress on implementing this?
